notebooks/WFC3/point_spread_function/hst_point_spread_function.ipynb

```diff
@@ -0,0 +1,1531 @@
+   "id": "30ec74bf-ee16-4829-82e8-eb56a3a4776b",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "os.chdir(data_dir)\n",
+    "!./../hst1pass.e HMIN=5 FMIN=1000.0 PMAX=66000.0 PERT=2 OUT=xympqks PSF=./../psfs/STDPSF_WFC3UV_F606W.fits id8048dyq_flc.fits\n",
+    "shutil.copy('id8048dyq_psf.fits', psfs_dir+'id8048dyq_psf_pert2.fits')\n",
+    "clear_output()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6b97646c-2d6a-4f61-ad12-8c3c915a368b",
+   "metadata": {},
+   "source": [
+    "Read in the perturbed model and examine the residuals."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b3cd48ed-a0ad-4e58-ad8b-4f297d65bbd9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "psf_model_perturbed_2 = data_dir+'id8048dyq_psf.fits'\n",
+    "model_uvis1 = GriddedPSFModel.read(psf_model_perturbed_2, format='stdpsf', detector_id=2)\n",
+    "# model_uvis2 = GriddedPSFModel.read(psf_model_perturbed_2, format='stdpsf', detector_id=1) # Would be used for UVIS 2.\n",
+    "photometry, psfphot = calculate_photometry(sci_data, sources, psf_model=model_uvis1)\n",
+    "resid = psfphot.make_residual_image(sci_data, (9, 9))\n",
+    "figure = plot_psf_results(sci_data, resid, 1818, 986, cutout_size)\n",
+    "save_figure(figure, plot_dir+'fig5.pdf', True)\n",
+    "calculate_residuals(resid, sources)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "14ba694b-ebe4-432b-9856-ff4ea9c9e060",
+   "metadata": {},
+   "source": [
+    "Interestingly, the residual flux is minutely larger when using the spatially-dependent model with this dataset, which can be seen by comparing the `flux_background_sub` values for each star in the above tables. This is primarily due to the fact that there are an insufficient number of bright, isolated stars to accurately constrain the model at each location on the detector. Thus, the best solution is produced in using `PERT=1` with hst1pass.\n",
+    "\n",
+    "In summary, the default models will work well for many science cases. If there are substantial residuals, then hst1pass can apply a perturbation that slightly changes the PSF model shape to compensate for different focus positions (`PERT=1`). It is also possible to account for both the focus and spatial variation of the PSF across the detector (`PERT=2`). However, changes in the PSF due to focus are generally more significant than the spatial variations unless sources are close to the edges of the detector, and accounting for both effects requires a very large number of bright, isolated stars.\n",
+    "\n",
+    "Finally, we rerun hst1pass with `PERT=1` so the best catalog can be used later. This concludes the section on using empirical PSF models provided by the WFC3 team."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "95ac8546-a490-40d9-a3db-6b1fe3d2d3dd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "os.chdir(data_dir)\n",
```

Comment:
Maybe instead of rerunning hst1pass (which could take some time depending on others computational resources), you could make save a copy of each pert results then overwrite the pert1 results to the desired name:
```
# after running hst1pass in 2.2.2
shutil.copy('id8048dyq_flc.xympqks', 'id8048dyq_flc_pert1.xympqks')
# after running hst1pass in 2.2.3 (for consistency)
shutil.copy('id8048dyq_flc.xympqks', 'id8048dyq_flc_pert2.xympqks')
# after plotting pert2 results in 2.2.3
shutil.copy('id8048dyq_flc_pert1.xympqks', 'id8048dyq_flc.xympqks')
```

Reply:
Okay, I've taken your advice here with a single tweak so users can more easily change catalogs, and so the main catalog is copied to the main directory for the repository so it's updated automatically. I updated the text:

"Finally, we copy the hst1pass catalog generated with PERT=1 so the best catalog is used later in the stacking analysis. We also copy the catalog to the main directory so users who are unable to run hst1pass can complete the tutorial. Users can change this cell to experiment with different catalogs made by running hst1pass with different settings."